### ISSUE TITLE
bug: Talkback focus does not get trapped on alert dialog

### ISSUE THREAD

**Issue by mastergogo** [ionic-team/ionic-framework#21946]
This is in relation to the issue #21598 that I had raised earlier about talkback focus not getting trapped in overlay components such as alert dialog or action sheets, if u keep navigating on the overlay components, talkback focus will go behind the overlay component, it should be trapped within the alert dialog unless the user closes it when the focus goes behind it can be very difficult for the screen reader user to navigate within the app.

Attached is the apk and steps to reproduce:

Turn on the talkback from accessibility settings and Launch the app:

1. Open any recipe out of two by double-tapping.
2. Navigate to delete button and double-tap to delete
3. An alert dialog will appear, keep navigating, or swipe forward using talkback till the focus reaches the "okay" button.
4. keep swiping and you will notice that focus will not be trapped within the alert and go behind instead.

Let me know if more info is needed. 
[app-debug.apk.zip](https://github.com/ionic-team/ionic-framework/files/5107021/app-debug.apk.zip)

**Comment by mastergogo** [comment]
Hi @liamdebeasi have you had the chance to check the above repo? Do i need to share anything else? let me know!

**Comment by mastergogo** [comment]
Hi @liamdebeasi, is the above repo helpful? Are you able to reproduce the issue? Let me know if I can help in anyway.

**Comment by liamdebeasi** [comment]
Thanks, I can reproduce the issue.

**Comment by mastergogo** [comment]
Hi @liamdebeasi any update on it?

**Comment by mastergogo** [comment]
Hi, any idea when this will be fixed?

**Comment by mastergogo** [comment]
Hi, @brandyscarney Is there a timeline on this by any chance? this issue keeps popping up and accessibility is at the core of our app and ADA compliance is very critical for our project. we would really appreciate it if this gets fixed at the earliest.

**Comment by kalpeshjain89** [comment]
Hi,
We are using Ionic framework to develop a mobile app which needs to be accessible. The App will cater to countries having accessibility compliance and laws in place thereby making A11y a prime requirement. Our launch time is around mid May, 2021. We would appreciate if anyone can give a TAT for fixing this issue?

**Comment by liamdebeasi** [comment]
Thanks for the issue and apologies for the delay. I am no longer able to reproduce this issue using the latest version of Ionic 6 and the code reproduction provided in https://github.com/ionic-team/ionic-framework/issues/21946#issuecomment-678342617. I recommend updating to the latest version of Ionic 6 to receive a fix for this bug.

I am going to close this, but please feel free to follow up if you have any additional questions.
